src/functions/orderBy.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+function compareAscending(value, other) {
+  if (value !== other) {
+    var valIsDefined = value !== undefined;
+    var valIsNull = value === null;
+
+    var othIsDefined = other !== undefined;
+    var othIsNull = other === null;
+
+    if (
+      (!othIsNull && value > other) ||
+      (valIsNull && othIsDefined) ||
+      !valIsDefined
+    ) {
+      return 1;
+    }
+    if (
+      (!valIsNull && value < other) ||
+      (othIsNull && valIsDefined) ||
+      !othIsDefined
+    ) {
+      return -1;
+    }
+  }
+  return 0;
+}
+
+/**
+ * @param {Array<object>} collection object with keys in attributes
+ * @param {Array<string>} iteratees attributes
+ * @param {Array<string>} orders asc | desc
+ */
+function orderBy(collection, iteratees, orders) {
+  if (!Array.isArray(collection)) {
+    return [];
+  }
+
+  if (!Array.isArray(orders)) {
+    orders = [];
+  }
+
+  var result = collection.map(function(value, index) {
+    return {
+      criteria: iteratees.map(function(iteratee) {
+        return value[iteratee];
+      }),
+      index: index,
+      value: value
+    };
+  });
+
+  result.sort(function comparer(object, other) {
+    var index = -1;
+
+    while (++index < object.criteria.length) {
+      var res = compareAscending(object.criteria[index], other.criteria[index]);
+      if (res) {
```

Comment:
We could change this block to 

```js
if (res) {
  if (orders[index] === 'desc') {
    return -res;
  } else {
    return res;
  }
}
```

If `index` is out of range, it will fail `=== 'desc'` anyway.

Reply:
I was thinking about that, but I think this was added in lodash to shortcut the loop, although reading it again, it might not make a difference

Reply:
hmmm no idea 🤷🏼‍♂️